### PR TITLE
Improve support >ES5 and Node.

### PR DESCRIPTION
**Pull request by NeilFraser**
Adds ‘yield’ and ‘**’ to JS order list.
Build generates files (more) parsable by Node.
This PR should have no effect on existing code.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,12 +42,9 @@
...
   },
   "dependencies": {
+    "google-closure-library": "^20171203.0.0",
     "install": "^0.8.8",
     "npm": "^4.4.4",
-    "closure-library": "^1.43629075.2",
     "webdriverio": "^4.6.2"
-  },
-  "devDependencies": {
-    "eslint": "2.9.0"
```
Comment (NeilFraser) [inline package.json]: Appears to be a duplicate 'devDependencies'.  Merging with the one above.

File: package.json [hunk 2]
```diff
@@ -41,12 +42,9 @@
     "unused": true
   },
   "dependencies": {
+    "google-closure-library": "^20171203.0.0",
     "install": "^0.8.8",
     "npm": "^4.4.4",
-    "closure-library": "^1.43629075.2",
```
Comment (NeilFraser) [inline package.json]: This is an abandoned unofficial 5 year old copy of Closure Library, not the real one.
https://github.com/gregrperkins/closure-library

File: generators/javascript.js [hunk 3]
```diff
@@ -84,8 +84,9 @@ Blockly.JavaScript.ORDER_TYPEOF = 4.5;         // typeof
 Blockly.JavaScript.ORDER_VOID = 4.6;           // void
 Blockly.JavaScript.ORDER_DELETE = 4.7;         // delete
 Blockly.JavaScript.ORDER_AWAIT = 4.8;          // await
-Blockly.JavaScript.ORDER_DIVISION = 5.1;       // /
-Blockly.JavaScript.ORDER_MULTIPLICATION = 5.2; // *
```
Comment (NeilFraser) [inline generators/javascript.js]: Mozilla now has / and * flipped on their page.  Makes no difference, but we might as well flip them accordingly.

File: build.py [hunk 4]
```diff
@@ -104,16 +104,10 @@ def run(self):
     f = open(self.target_filename, 'w')
     f.write(HEADER)
     f.write("""
-var isNodeJS = !!(typeof module !== 'undefined' && module.exports &&
-                  typeof window === 'undefined');
+this.IS_NODE_JS = !!(typeof module !== 'undefined' && module.exports);
 
-if (isNodeJS) {
-  var window = {};
```
Comment (NeilFraser) [inline build.py]: Creating a fake global window object for Node is bad programming.

File: build.py [hunk 5]
```diff
@@ -104,16 +104,10 @@ def run(self):
     f = open(self.target_filename, 'w')
     f.write(HEADER)
     f.write("""
-var isNodeJS = !!(typeof module !== 'undefined' && module.exports &&
```
Comment (NeilFraser) [inline build.py]: Creating a global variable is bad.  Make a property which can be deleted later.

File: package.json [hunk 6]
```diff
@@ -24,7 +24,8 @@
   "license": "Apache-2.0",
   "private": true,
   "devDependencies": {
-    "jshint": "latest"
+    "jshint": "latest",
+    "eslint": "2.9.0"
```
Comment (AnmAtAnm) [inline package.json]: Aren't we in the process of moving this to 4.x from 3.x?

Comment (rachel-fenichel) [reply package.json]: Apparently from 2.9.0 -> 4.16.0.  But I'll handle that/any conflicts from this.

**Pull request merged**